metadata-ingestion/src/datahub/ingestion/source/fs/fs_base.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+from typing import Iterable
+from urllib import parse
+from abc import ABCMeta, abstractmethod
+
+
+@dataclass
+class FileStatus:
+    path: str
+    size: int
+    is_file: bool
+
+    def __str__(self):
+        return f"FileStatus({self.path}, {self.size}, {self.is_file})"
+
+
+class FileSystem(metaclass=ABCMeta):
+
+    @classmethod
+    def create_fs(cls) -> "FileSystem":
+        raise NotImplementedError('File system implementations must implement "create_fs"')
+
+    @abstractmethod
+    def open(self, path: str, **kwargs):
```

Comment:
is it possible to add a return type annotation here, or is it too messy?

Reply:
I think at this point we don't know actual return type and it depends on underlying implementation. **smart_open** says that **open** method returns **A file-like object**